packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -79,6 +83,67 @@ export const Table = memo((props: Props) => {
     return EXTENDED_ROW_HEIGHT;
   }, [footerItems]);
 
+  // Create a color map to color by row
+  const colorMap: RowToColor = {};
+
+  data.fields.map((field) => {
+    // Get threshold configurations for this field
+    const { thresholds } = field.config;
+
+    // If we have no thresholds we have nothing to do
+    if (thresholds === undefined) {
+      return;
+    }
+
+    for(let i = 0; i < field.values.length; i++) {
+      const value = field.values[i];
+
+      // If we're not dealing with numbers we
+      // don't need to do anything
+      if (field.type !== FieldType.number) {
+        continue;
+      }
+
+      for (let j = (thresholds.steps.length - 1); j >= 0; j--) {
+        const threshold = thresholds.steps[j];
+    
+        // TODO: Currently works when there's one numeric field, need
+        // to consider the case of multiple numeric fields, may 
+        // need to use field overrides but 
+        // not sure about the feasability of that approach
+        if (value >= threshold.value && threshold.value !== (-Infinity)) {
+          if (colorMap[i] === undefined) {
+            colorMap[i] = threshold.color;
+            break;
+          }
+        }  
+      }
```

Comment:
@codeincarnate 

Why just thresholds?? we have many colors schemes in grafana. I am really confusing by this . 

The problem seems pretty simple. Have the color of one of the fields apply to the row. Why introducing all this custom rowToColor and thresholds logic??

https://github.com/grafana/grafana/pull/83247

Reply:
In terms of the thresholds, it's been one of the use cases that's been asked for (coloring beyond a given threshold). It's an initial stab at it, and wasn't sure about this route honestly :)

Have a design doc fleshing out more (including what you mentioned) here https://docs.google.com/document/d/1en6BS3V9qrjmo6sVQYWLdZQibGSaLWSnYSC_Y779Mrw/edit